plugin.cpp

```diff
@@ -82,44 +91,49 @@ void setPluginConfig(Random *random, ConfigCategory *config)
 	if (config->itemExists("asset"))
 		random->setAssetName(config->getValue("asset"));
 
-	unsigned int nAssets = stoul(config->getValue("numAssets"), nullptr, 0);
-	if (nAssets <= 0)
+	unsigned int nAssets = 1;
+	if (config->itemExists("numAssets"))
+	{
+		nAssets = stoul(config->getValue("numAssets"), nullptr, 0);
+		if (nAssets <= 0)
+		{
+			throw runtime_error("The value of numAssets, number of unique assets to simulate, must be greater than 0");
+		}
+	}
+	random->setNumAssets(nAssets);
+
+	unsigned int numValues = 1;
+	if (config->itemExists("numValues"))
 	{
-		throw runtime_error("The value of numAssets must be greater than 0");
+		numValues = stoul(config->getValue("numValues"), nullptr, 0);
+		if (numValues <= 0)
+		{
+			throw runtime_error("The value of numValues, number of values to be returned per poll call, must be greater than 0");
```

Comment:
Throwing exception here works as expected?

Reply:
Well this should be handled by south service. However, since we already have minimum limit set in config this code  will not be reached. I used it to keep the code in synch with existing code.